### PR TITLE
ddl, stat: switch to new struct for add/truncate/drop partition

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve?

Issue Number: ref #55723

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/statistics/handle/ddl/truncate_partition.go [hunk 1]
```diff
@@ -16,6 +16,7 @@ package ddl
 
 import (
 	"github.com/pingcap/errors"
+	util2 "github.com/pingcap/tidb/pkg/ddl/util"
```
Comment (hawkingrei) [inline pkg/statistics/handle/ddl/truncate_partition.go:19]:
```suggestion
	ddlutil "github.com/pingcap/tidb/pkg/ddl/util"
```

**Comment by lance6716** [comment]
ptal @hawkingrei

File: pkg/ddl/util/schema_change_notifier.go [hunk 2]
```diff
@@ -101,3 +103,74 @@ func (s *SchemaChangeEvent) GetTruncateTableInfo() (
 	intest.Assert(s.tp == model.ActionTruncateTable)
 	return s.newTableInfo, s.oldTableInfo
 }
+
+// NewAddPartitionEvent creates a SchemaChangeEvent whose type is
```
Comment (fzzf678) [inline pkg/ddl/util/schema_change_notifier.go:107]: The comment seem repeated

**Review by Rustin170506 — APPROVED** [review]
Looks good to me. Thanks! 👍

**Review by fzzf678 — APPROVED** [review]
rest lgtm

File: pkg/ddl/util/schema_change_notifier.go (hunk 2 above)
Comment (lance6716) [reply pkg/ddl/util/schema_change_notifier.go:107]: I'll remove it in next PR

**Pull request merged**